### PR TITLE
iX for listing .c found with dbginfo #6159

### PR DESCRIPTION
**Pull request by Maijin**

### CODE REVIEW

**Comment by Maijin** [comment]
https://github.com/radare/radare2-regressions/pull/1413

File: libr/core/cbin.c [hunk 1]
```diff
@@ -1020,6 +1020,38 @@ static int bin_pdb(RCore *core, int mode) {
...
+	RBinFile * binfile = r->bin->cur;
+	SdbListIter *iter;
+	RListIter *iter2;
+	char* srcline;
+	SdbKv *kv;
+	SdbList *ls = sdb_foreach_list (binfile->sdb_addrinfo, false);
+	ls_foreach (ls, iter, kv) {
+		char *v = kv->value;
+		RList *list = r_str_split_list (v, "|");
+		if (!strstr (r_list_get_bottom (list), "0x")){
```
Comment (xarkes) [inline libr/core/cbin.c]: If for some reason the list has no head, r_list_get_bottom might return NULL. Better use it appart and check its value.

File: libr/core/cbin.c [hunk 2]
```diff
@@ -1020,6 +1020,38 @@ static int bin_pdb(RCore *core, int mode) {
...
+			r_list_append (final_list, r_list_get_bottom (list)); // Should check if already here and discard
```
Comment (xarkes) [inline libr/core/cbin.c]: Same here. Otherwise someday someone will cry because of segfaults.

Comment (xarkes) [reply libr/core/cbin.c]: Plus it avoids calling twice the same function.

**Pull request merged**